NppExec_parse_help_topics.py

```diff
@@ -0,0 +1,71 @@
+import sys
+import re
+import json
+import urllib.parse
+from html.parser import HTMLParser
+from pathlib import Path, PurePath
+
+class NppExecDocParser(HTMLParser):
+    CHARSET = 'cp1252'
+
+    def __init__(self):
+        super().__init__()
+        self.topics = {}
+        self.doc_title = ''
+        self.is_doc_title, self.is_content = False, False
+
+    def handle_starttag(self, tag: str, attrs: str):
+        _tag = tag.lower()
+        self.is_doc_title = (_tag == 'title')
+        self.is_content = \
+            re.match(r'h[1-6]', _tag) is not None or \
+            _tag in ['blockquote', 'body', 'div', 'p', 'pre']
+
+    def handle_data(self, data: str):
+        if self.is_doc_title:
+            self.doc_title = data.strip()
+            self.topics[self.doc_title] = { 'keywords': [] }
+            self.is_doc_title = False
+        elif self.is_content and bool(data.strip()):
+            try:
+                words = data.strip().split()
+                self.topics[self.doc_title]['keywords'].extend([w.replace("\"",'').replace("\'",'') for w in words])
+            finally:
+                self.is_content = False
+
+    def collect_info(self, doc: Path):
+        try:
+            with open (doc, 'r', encoding=self.CHARSET) as d:
+                self.feed(d.read())
+
+            self.topics[self.doc_title]['path'] = "./{0}".format(urllib.parse.quote(doc.name, safe='./:'))
+        except (KeyError, AttributeError, IOError) as e:
+            print(repr(e), file=sys.stderr)
+            sys.exit(2)
+
+
+def run():
+    parser = NppExecDocParser()
+    docdir = Path(PurePath(sys.path[0]) / 'docs' / 'NppExec_Manual')
+    topics = Path(PurePath(sys.path[0]) / 'docs' / 'topics.js')
+
+    if not docdir.exists():
+        print('No such directory:', docdir)
+        sys.exit(1)
+
+    print('Parsing docs in', docdir, '...')
+
+    for doc in docdir.glob('**/*.html'):
+        if doc.stem in ['toc', 'index']:
+            continue
+
+        parser.collect_info(doc)
+
+    print('Found', len(parser.topics.keys()), 'topics')
+
+    with open (topics, 'w', encoding=NppExecDocParser.CHARSET) as out:
+        out.write("var NPPEXEC_HELP_TOPICS = {0}".format(json.dumps(parser.topics)))
```

Comment:
suggesting `json.dumps(parser.topics, indent=2))` for easier navigation through the generated file 'topics.js' by eyes

Reply:
Web scripts are conventionally in "minified" form, but you're right that an editable format is better.

You could always devise a "production" workflow to minify everything before publishing the docs. Replacing every match on `/(?<=[{}:;,])\s+/` with a `null` string would be good enough, I think.

Reply:
Or, even better, there can be a command-line argument (e.g. using `argparse`) to specify whether we want a human-readable output or not

Reply:
Added argument parsing in https://github.com/d0vgan/nppexec/commit/58616972ea26cf0a4a189120f252e2e5452e1f74